staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -194,11 +194,28 @@ func (d *Helper) DeleteOrEvictPods(pods []corev1.Pod) error {
 
 func (d *Helper) evictPods(pods []corev1.Pod, policyGroupVersion string, getPodFn func(namespace, name string) (*corev1.Pod, error)) error {
 	returnCh := make(chan error, 1)
-
+	// 0 timeout means infinite, we use MaxInt64 to represent it.
+	var globalTimeout time.Duration
+	if d.Timeout == 0 {
+		globalTimeout = time.Duration(math.MaxInt64)
+	} else {
+		globalTimeout = d.Timeout
+	}
+	ctx, cancel := context.WithTimeout(context.TODO(), globalTimeout)
+	defer cancel()
 	for _, pod := range pods {
 		go func(pod corev1.Pod, returnCh chan error) {
 			for {
 				fmt.Fprintf(d.Out, "evicting pod %q\n", pod.Name)
+				select {
+				case <-ctx.Done():
+					// return here or we'll leak a goroutine.
+					returnCh <- fmt.Errorf("error when evicting pod %q: global timeout", pod.Name)
+					return
+				default:
+					// break the select.
```

Comment:
do we need this or is it in a style guide somewhere?

Reply:
```
A select blocks until one of its cases can run, then it executes that case. It chooses one at random if multiple are ready. 
```
https://tour.golang.org/concurrency/5

Yeah, looks like select blocks until one of it's cases run.  Probably don't need 'break' specifically.

@pwittrock

Reply:
yes, my question was about the break.  since the break is a no-op it is confusing and looked like a possible programming error

Reply:
Yeah, okay, not needed and removed, thanks.